main.tf

```diff
@@ -1,46 +1,68 @@
 locals {
-  datacenters     = join(",", var.nomad_datacenters)
-  buckets         = var.hive_bucket # output variable for presto
-  hive_env_vars   = join("\n",
+  datacenters   = join(",", var.nomad_datacenters)
+  buckets       = var.hive_bucket # output variable for presto
+  hive_env_vars = join("\n",
     concat([
       "JUST_EXAMPLE_ENV=some-value",
     ], var.hive_container_environment_variables)
   )
+  vault_provider = var.postgres_vault_secret.use_vault_provider || var.minio_vault_secret.use_vault_provider
+  vault_kv_policy_name = jsonencode(
+    concat(
+      [var.postgres_vault_secret.vault_kv_policy_name],
+      [var.minio_vault_secret.vault_kv_policy_name]
+    )
+  )
 }
 
 data "template_file" "template_nomad_job_hive" {
 
   template = file("${path.module}/conf/nomad/hive.hcl")
 
-  vars      = {
-    use_canary                = var.use_canary
-    service_name              = var.hive_service_name
-    datacenters               = local.datacenters
-    namespace                 = var.nomad_namespace
+  vars = {
+    use_canary   = var.use_canary
+    service_name = var.hive_service_name
+    datacenters  = local.datacenters
+    namespace    = var.nomad_namespace
 
-    local_docker_image        = var.local_docker_image
-    image                     = var.hive_docker_image # !NB: no affect when `local_docker_image=true`
-    port                      = var.hive_container_port
-    envs                      = local.hive_env_vars
+    use_vault_provider    = local.vault_provider
+    vault_kv_policy_name  = local.vault_kv_policy_name
 
-    cpu                       = var.resource.cpu
-    memory                    = var.resource.memory
+    local_docker_image = var.local_docker_image
+    image              = var.hive_docker_image # !NB: no affect when `local_docker_image=true`
```

Comment:
Just want to say I really like small additions like this in the code! Makes everything a lot more readable, which is important if people are going to start using our code. Especially now that they are starting to become quite big cobwebs

Reply:
I put some more comments to explicitly mark ignored variables